types/src/on_chain_config/execution_config.rs

```diff
@@ -132,13 +146,31 @@ pub enum TransactionDeduperType {
 pub enum BlockGasLimitType {
     NoLimit,
     Limit(u64),
+    LimitWithConflictAware {
+        block_gas_limit: u64,
+        conflict_overlap_window: u32,
```

Comment:
I might consider calling it `conflict_detection_window` or `conflict_penalty_window`